### PR TITLE
Compacter dictionary with Scopus journal names

### PR DESCRIPTION
**Pull request by a-fent**
A somewhat compacter dictionary with additional journal titles, mostly compiled from the previously used sources:
- duplicate entries in the same mode are not repeated
- American surnames with <0.001 are not included (by my estimation this means <3,000 US population)
- Additional journal names based on the publications indexed by Scopus

Happy to share the simple script used to compile the dictionary file.

### CODE REVIEW

**Comment by inukshuk** [comment]
First off, this is great!

@namyra and I have also been working on improving the dictionary recently along with other updates that have been going on. We're planning to set up a separate Gem/Repository at https://github.com/inukshuk/anystyle-data -- to make it easier to update the dictionary separately from the parser (and not consume unecessary disk space if you have different versions of the parser gem installed). We'll incorporate the data there.

Also, a heads up that the refactoring of the parser is approaching a point where it is useful again. I've improved the model (patterns) a lot (I hope!) and we've added tooling to make it much easier to compare results and manipulate training data. If you're working on the parser, would you like me to give you a quick summary of the changes? (Or have you been following along anyway?)

**Comment by a-fent** [comment]
Ah, great that you've already set up a repo for the dictionary data. Is it useful for me to submit my local data (e.g. Scopus) and the script for compiling dict.txt.gz out of multiple text sources there?

I am really impressed with the recent changes, which I just merged into my local tree this morning. I am getting very good accuracy even on quite difficult and heterogenous real data. A quick summary of the main changes would be interesting (so far I see: changes to patterns, real XML training data)

Out of interest, are you developing the model specification with a formal test & training set?

**Comment by inukshuk** [comment]
Yes, we'll want to add a rake task in `anystyle-data` to compile the dictionary out of several files; that will make maintenance much easier hopefully. PRs welcome over there of course.

As a quick summary, what I'm currently working on is improving the parser architecture to allow us to add the 'finder' component and share most of the utility functionality (like loading and comparing training data) between the two; also improving feature and normalizer code as I work my way through the code base.

The most interesting new feature so far was actually moved to the `Wapiti`: the `Dataset`, `Sequence` and `Token` classes. Basically, they help me create, compare, load, save, convert, etc. input and training data sets. Because this is pretty much in flux, here's just a quick example:

```
# Load the core training set
>> c = Wapiti::Dataset.open './res/core.xml'
=> #<Wapiti::Dataset sequences={681}>

# Load another training set I'm working with. Data sets can be tagged or untagged; XML
# sets are tagged by definition.
>> s = Wapiti::Dataset.open './res/stanford.xml'
=> #<Wapiti::Dataset sequences={500}>

# Label the set s: this returns a new data set
>> AnyStyle.parser.label(s)
=> #<Wapiti::Dataset sequences={500}>

# You can compare the new and original data set to get only
# the references which were tagged differently. Here we get
# a data set with the labels the parser predicted; if we change
# the order we get the labels from the original training set instead.
>> x = AnyStyle.parser.label(s) - s
=> #<Wapiti::Dataset sequences={164}>

# Each data set can be converted to a different format. You can combine
# multiple sets using set operations or save them to disk. As an example,
# here I take two random sequences out of the set and print them as XML.
>> puts x.sample(2).to_xml(indent: 2).to_s
<?xml version="1.0" encoding="UTF-8"?>
<dataset>
  <sequence>
    <citation-number>[175]</citation-number>
    <author>M. Vollmer, E. J. Mayer, W. W. Rühle, A. Kurtenbach, and K. Eberl,</author>
    <title>“Exci- ton relaxation dynamics in quantum dots with strong confinement,”</title>
    <journal>Physical Review B 54,</journal>
    <volume>R17292–R17295</volume>
    <date>(1996).</date>
  </sequence>
  <sequence>
    <citation-number>[176]</citation-number>
    <author>K. Eberl, A. Kurtenbach, K. Häusler, F. Noll, and W. W. Rühle,</author>
    <title>“Preparation and Time Resolved Photoluminescence of Nanoscale InP Islands in In0. 48Ga0. 52P,”</title>
    <booktitle>in “MRS Proceedings,” ,</booktitle>
    <volume>vol. 379</volume>
    <date>(1995),</date>
    <volume>vol. 379.</volume>
  </sequence>
</dataset>
```

There's more going on, but if you're working with the parser the `Dataset` class is extremely powerful, so I wanted to give you a heads up.

With your last question, do you mean what wapiti calls a 'development' set (used for the stop criteria)? I'm not doing that at the moment, instead using all the training data and compacting the model at the end. But I'm still experimenting with incrementally building up models so I using a development set is something that has been on my mind.

**Comment by a-fent** [comment]
I rolled up the content changes with a bit of Rakefile to compile dict.txt.gz and submitted it as https://github.com/inukshuk/anystyle-data/pull/1

That PR assumes that the small change to dictionary.rb (setting the mode by writing @mode@ when reading) is OK and applied in this tree. The dict.txt.gz bit of this patch is no longer needed.

**Comment by a-fent** [comment]
Thank you very much for the summary of the changes. I had seen that proper XML was being used but hadn't see how it was within a proper Dataset concept. Being able to compare traininq sequences with parsed training sets is really exciting - something I will want to use and what I was thinking of when I asked about developing the model. Looking forward to it!

**Comment by a-fent** [comment]
This is now merged onto anystyle-data or superseded, so I'm closing.

Sylvester, your note here https://github.com/inukshuk/anystyle/pull/89#issuecomment-356436735 might be interesting to at some point move into the Wapiti documentation, so I've saved it on my tree.
